### PR TITLE
[FL-1926] Flipper File Format addons

### PR DESCRIPTION
**Pull request by DrZlo13**
# What's new

- Misc addons to library

# Verification 

- Run unit tests

# Checklist (do not modify)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: lib/flipper-file/file-tool.c [hunk 1]
```diff
@@ -0,0 +1,213 @@
+#include "file-tool.h"
+
+const char flipper_file_eoln = '\n';
+const char flipper_file_eolr = '\r';
+
+#ifndef MIN
```
Comment (skotopes) [inline lib/flipper-file/file-tool.c]: already in furi core

File: lib/flipper-file/flipper-file-string.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return result;
+}
+
+bool flipper_file_write_string(FlipperFile* flipper_file, const char* key, string_t data) {
+    furi_assert(flipper_file);
+    return flipper_file_write_string_internal(flipper_file->file, key, data, 0);
+}
+
+bool flipper_file_write_string_cstr(FlipperFile* flipper_file, const char* key, const char* data) {
+    bool result = false;
```
Comment (skotopes) [inline lib/flipper-file/flipper-file-string.c]: Asserting one places and ignoring another

File: lib/flipper-file/file-tool.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+extern const char flipper_file_eoln;
+extern const char flipper_file_eolr;
+
+/**
+ * Negative seek helper
+ * @param file 
+ * @param offset 
+ * @return bool 
+ */
+bool file_tool_seek(File* file, int32_t offset);
```
Comment (skotopes) [inline lib/flipper-file/file-tool.h]: File tool is quite strange name, file helper is more common and suitable

File: lib/flipper-file/flipper-file-tool.c [hunk 4]
```diff
@@ -0,0 +1,118 @@
+#include "flipper-file-tool.h"
```
Comment (skotopes) [inline lib/flipper-file/flipper-file-tool.c]: how about to start to use `_` )

**Review by skotopes — CHANGES_REQUESTED** [review]

File: lib/flipper-file/flipper-file-string.c (hunk 2 above)
Comment (DrZlo13) [reply lib/flipper-file/flipper-file-string.c]: flipper_file_write_string_cstr calls flipper_file_write_string, which already contains assert

File: lib/flipper-file/file-tool.c (hunk 1 above)
Comment (DrZlo13) [reply lib/flipper-file/file-tool.c]: I cannot find anything related

File: lib/flipper-file/file-tool.h (hunk 3 above)
Comment (DrZlo13) [reply lib/flipper-file/file-tool.h]: Fixed in 0a72f6e53cfd9b6cce00573fb7805bd9534edd89

File: lib/flipper-file/flipper-file-tool.c (hunk 4 above)
Comment (DrZlo13) [reply lib/flipper-file/flipper-file-tool.c]: Fixed in 0a72f6e

File: lib/flipper-file/file-tool.c (hunk 1 above)
Comment (DrZlo13) [reply lib/flipper-file/file-tool.c]: Comment was related to MIN. Fixed.

**Review by skotopes — APPROVED** [review]

**Pull request merged**